doi-transformer/src/test/java/no/unit/nva/doi/transformer/model/crossrefmodel/CrossrefDocumentTest.java

```diff
@@ -30,4 +31,20 @@ public void testSettersAndGetters() throws IOException {
         assertThat(crossRefResponse, doesNotHaveNullOrEmptyFields());
         assertThat(serializedAgain, is(equalTo(serializedObject)));
     }
+
+    @Test
+    public void testSettersAndGettersCrossrefBook() throws IOException {
+
+        String resourceString = IoUtils.stringFromResources(CROSSREF_BOOK_RESOURCE);
+        CrossrefApiResponse crossRefResponse = objectMapper.readValue(resourceString, CrossrefApiResponse.class);
+        String serializedObject = objectMapper.writeValueAsString(crossRefResponse);
+        CrossrefApiResponse deserializedObject = objectMapper.readValue(serializedObject, CrossrefApiResponse.class);
+        //cannot test deep equality on those classes. we compare therefore strings.
+        String serializedAgain = objectMapper.writeValueAsString(deserializedObject);
```

Comment:
Convert to JsonNodes and compare those.

Reply:
JsonNode it is...

Reply:
use objectMapper.convertValue(deserialzedObject,JsonNode.class) to keep the assertion line clean